File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> participantCount = new HashMap<>();

        for (String name : participant) {
            participantCount.put(name, participantCount.getOrDefault(name, 0) + 1);
        }

        for (String name : completion) {
            participantCount.put(name, participantCount.get(name) - 1);
        }

        for (Map.Entry<String, Integer> entry : participantCount.entrySet()) {
            if (entry.getValue() != 0) {
                return entry.getKey();
            }
        }

        return "";
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    for (const player of participant) {
        map.set(player, (map.get(player) || 0) + 1);
    }
    for (const player of completion) {
        map.set(player, map.get(player) - 1);
    }
    for (const [name, count] of map) {
        if (count > 0) return name;
    }
}